test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsSolutionRestoreServiceTests.cs

```diff
@@ -2387,6 +2388,34 @@ public void ToPackageSpec_TwoProjectReferencesToSameProject_DeduplicatesProjectR
         private delegate void TryGetProjectNamesCallback(string projectPath, out ProjectNames projectNames);
         private delegate bool TryGetProjectNamesReturns(string projectPath, out ProjectNames projectNames);
 
+        [Fact]
+        public void ToPackageSpec_PackageDownloadWithNoVersion_ThrowsException()
+        {
+            // Arrange
+            string packageName = "package";
+            ProjectNames projectName = new ProjectNames(@"n:\path\to\current\project.csproj", "project", "project.csproj", "project", Guid.NewGuid().ToString());
+            var emptyReferenceItems = Array.Empty<VsReferenceItem>();
+            var targetFrameworks = new VsTargetFrameworkInfo2[]
+            {
+                new VsTargetFrameworkInfo2("net5.0",
+                    packageReferences: emptyReferenceItems,
+                    projectReferences: emptyReferenceItems,
+                    packageDownloads: new List<IVsReferenceItem>
+                    { new VsReferenceItem(packageName, new VsReferenceProperties(new []
+                    {
+                        new VsReferenceProperty("Version", null)
+                    }))
+                    },
+                    frameworkReferences: emptyReferenceItems,
+                    projectProperties: Array.Empty<IVsProjectProperty>())
+            };
```

Comment:
Indentation is off.  Also, curly braces should both be on their own line or on the same line (if the body is small).

```suggestion
            {
                new VsTargetFrameworkInfo2(
                    targetFrameworkMoniker: "net5.0",
                    packageReferences: emptyReferenceItems,
                    projectReferences: emptyReferenceItems,
                    packageDownloads: new List<IVsReferenceItem>
                    {
                        new VsReferenceItem(
                            packageName,
                            new VsReferenceProperties(new [] { new VsReferenceProperty("Version", value: null) }))
                    },
                    frameworkReferences: emptyReferenceItems,
                    projectProperties: Array.Empty<IVsProjectProperty>())
            };
```